### DISCUSSION TITLE
Improvement for the VT communication

### DISCUSSION THREAD

**Discussion by thiagozf1** [Open-Agriculture/AgIsoStack-plus-plus#298] [Ideas]
Hi guys, today I'm here just to make a suggestion. In my application for the stack I have to send a lot of messages to the VT and sometimes they aren't properly received by it. So I've added some code to not send any messages before receiving a response from the VT (it was specialy usefull with `send_change_stringvalue` as it is a somewhat long message) and than it worked fine. So I was thinking that it would be nice (if possible) to add something like this to the stack. And that is it, bye and thanks for all the assistance you guys have been giving me!

**Comment by ad3154** [comment]
Hmm, that's a pretty good idea. It would require us to queue messages in the VT client or to check against some kind of lock to make sure we return false on the transmits until a response is received or a message times out. Maybe both.
